Cannot build Benchmarks

Building src/Benchmarks errors with unable to find a 9.0 version of Microsoft.Extensions.Logging 

```
❯ ~/.dotnet/dotnet --info
.NET SDK:
 Version:           9.0.100-rc.2.24474.11
 Commit:            315e1305db
 Workload version:  9.0.100-manifests.cea95dba
 MSBuild version:   17.12.0-preview-24473-03+fea15fbd1

Runtime Environment:
 OS Name:     ubuntu
 OS Version:  22.04
 OS Platform: Linux
 RID:         linux-x64
 Base Path:   /data_sdb/alahay01/.dotnet/sdk/9.0.100-rc.2.24474.11/

.NET workloads installed:
There are no installed workloads to display.
Configured to use loose manifests when installing new manifests.

Host:
  Version:      9.0.0-rc.2.24473.5
  Architecture: x64
  Commit:       990ebf52fc

.NET SDKs installed:
  6.0.405 [/data_sdb/alahay01/.dotnet/sdk]
  7.0.102 [/data_sdb/alahay01/.dotnet/sdk]
  9.0.100-rc.2.24474.11 [/data_sdb/alahay01/.dotnet/sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 6.0.13 [/data_sdb/alahay01/.dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 7.0.2 [/data_sdb/alahay01/.dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 9.0.0-rc.2.24474.3 [/data_sdb/alahay01/.dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 6.0.13 [/data_sdb/alahay01/.dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.2 [/data_sdb/alahay01/.dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 9.0.0-rc.2.24473.5 [/data_sdb/alahay01/.dotnet/shared/Microsoft.NETCore.App]
```

```
❯ git clone https://github.com/aspnet/Benchmarks.git
❯ cd Benchmarks/src/Benchmarks
```

```
❯ ~/.dotnet/dotnet build -c Release
Welcome to .NET 9.0!
---------------------
SDK Version: 9.0.100-rc.2.24474.11

Telemetry
---------
The .NET tools collect usage data in order to help us improve your experience. It is collected by Microsoft and shared with the community. You can opt-out of telemetry by setting the DOTNET_CLI_TELEMETRY_OPTOUT environment variable to '1' or 'true' using your favorite shell.

Read more about .NET CLI Tools telemetry: https://aka.ms/dotnet-cli-telemetry

----------------
Installed an ASP.NET Core HTTPS development certificate.
To trust the certificate, run 'dotnet dev-certs https --trust'
Learn about HTTPS: https://aka.ms/dotnet-https

----------------
Write your first app: https://aka.ms/dotnet-hello-world
Find out what's new: https://aka.ms/dotnet-whats-new
Explore documentation: https://aka.ms/dotnet-docs
Report issues and find source on GitHub: https://github.com/dotnet/core
Use 'dotnet --help' to see available commands or visit: https://aka.ms/dotnet-cli
--------------------------------------------------------------------------------------
    /data_sdb/alahay01/foo/Benchmarks/src/Benchmarks/Benchmarks.csproj : error NU1102:
      Unable to find package Microsoft.Extensions.Logging with version (>= 9.0.0)
        - Found 2071 version(s) in dotnet6 [ Nearest version: 6.0.2-mauipre.1.22102.15 ]
        - Found 1562 version(s) in dotnet7 [ Nearest version: 7.0.0-rtm.22511.4 ]
        - Found 1459 version(s) in dotnet8 [ Nearest version: 8.0.0-rtm.23523.3 ]
        - Found 1414 version(s) in dotnet9 [ Nearest version: 9.0.0-rtm.24516.5 ]
        - Found 126 version(s) in dotnet-public [ Nearest version: 9.0.0-rc.2.24473.5 ]
        - Found 1 version(s) in dotnet7-transport [ Nearest version: 7.0.0-rc.1.22426.10 ]
        - Found 0 version(s) in dotnet-eng
        - Found 0 version(s) in dotnet10-transport
        - Found 0 version(s) in dotnet6-transport
        - Found 0 version(s) in dotnet8-transport
        - Found 0 version(s) in dotnet9-transport
    /data_sdb/alahay01/foo/Benchmarks/src/Benchmarks/Benchmarks.csproj : error NU1102:
      Unable to find package System.ServiceProcess.ServiceController with version (>= 9.0.0)
        - Found 2069 version(s) in dotnet6 [ Nearest version: 6.0.2-mauipre.1.22102.15 ]
        - Found 1560 version(s) in dotnet7 [ Nearest version: 7.0.0-rtm.22511.4 ]
        - Found 1458 version(s) in dotnet8 [ Nearest version: 8.0.0-rtm.23523.3 ]
        - Found 1414 version(s) in dotnet9 [ Nearest version: 9.0.0-rtm.24516.5 ]
        - Found 94 version(s) in dotnet-public [ Nearest version: 9.0.0-rc.2.24473.5 ]
        - Found 1 version(s) in dotnet-eng [ Nearest version: 5.0.0-alpha.1.19618.1 ]
        - Found 1 version(s) in dotnet7-transport [ Nearest version: 7.0.0-rc.1.22426.10 ]
        - Found 0 version(s) in dotnet10-transport
        - Found 0 version(s) in dotnet6-transport
        - Found 0 version(s) in dotnet8-transport
        - Found 0 version(s) in dotnet9-transport
    /data_sdb/alahay01/foo/Benchmarks/src/Benchmarks/Benchmarks.csproj : error NU1103:
      Unable to find a stable package Microsoft.Extensions.FileProviders.Embedded with version (>= 9.0.0)
        - Found 2100 version(s) in dotnet6 [ Nearest version: 6.0.0-rtm.21520.4 ]
        - Found 1621 version(s) in dotnet7 [ Nearest version: 7.0.0-rtm.22513.3 ]
        - Found 1593 version(s) in dotnet8 [ Nearest version: 9.0.0-alpha.1.23417.3 ]
        - Found 1148 version(s) in dotnet9 [ Nearest version: 10.0.0-alpha.1.24416.1 ]
        - Found 198 version(s) in dotnet-public [ Nearest version: 9.0.0-rc.2.24474.3 ]
        - Found 1 version(s) in dotnet7-transport [ Nearest version: 7.0.0-rc.1.22427.2 ]
        - Found 0 version(s) in dotnet-eng
        - Found 0 version(s) in dotnet10-transport
        - Found 0 version(s) in dotnet6-transport
        - Found 0 version(s) in dotnet8-transport
        - Found 0 version(s) in dotnet9-transport
    /data_sdb/alahay01/foo/Benchmarks/src/Benchmarks/Benchmarks.csproj : error NU1102:
      Unable to find package Microsoft.Extensions.Logging.Console with version (>= 9.0.0)
        - Found 2071 version(s) in dotnet6 [ Nearest version: 6.0.2-mauipre.1.22102.15 ]
        - Found 1562 version(s) in dotnet7 [ Nearest version: 7.0.0-rtm.22511.4 ]
        - Found 1459 version(s) in dotnet8 [ Nearest version: 8.0.0-rtm.23523.3 ]
        - Found 1414 version(s) in dotnet9 [ Nearest version: 9.0.0-rtm.24516.5 ]
        - Found 126 version(s) in dotnet-public [ Nearest version: 9.0.0-rc.2.24473.5 ]
        - Found 1 version(s) in dotnet7-transport [ Nearest version: 7.0.0-rc.1.22426.10 ]
        - Found 0 version(s) in dotnet-eng
        - Found 0 version(s) in dotnet10-transport
        - Found 0 version(s) in dotnet6-transport
        - Found 0 version(s) in dotnet8-transport
        - Found 0 version(s) in dotnet9-transport
    /data_sdb/alahay01/foo/Benchmarks/src/Benchmarks/Benchmarks.csproj : error NU1102:
      Unable to find package Microsoft.Extensions.DependencyInjection with version (>= 9.0.0)
        - Found 2071 version(s) in dotnet6 [ Nearest version: 6.0.2-mauipre.1.22102.15 ]
        - Found 1562 version(s) in dotnet7 [ Nearest version: 7.0.0-rtm.22511.4 ]
        - Found 1459 version(s) in dotnet8 [ Nearest version: 8.0.0-rtm.23523.3 ]
        - Found 1414 version(s) in dotnet9 [ Nearest version: 9.0.0-rtm.24516.5 ]
        - Found 126 version(s) in dotnet-public [ Nearest version: 9.0.0-rc.2.24473.5 ]
        - Found 1 version(s) in dotnet7-transport [ Nearest version: 7.0.0-rc.1.22426.10 ]
        - Found 0 version(s) in dotnet-eng
        - Found 0 version(s) in dotnet10-transport
        - Found 0 version(s) in dotnet6-transport
        - Found 0 version(s) in dotnet8-transport
        - Found 0 version(s) in dotnet9-transport

Restore failed with 5 error(s) in 18.1s
```

--- a/src/Benchmarks/Middleware/FortunesDapperMiddleware.cs
+++ b/src/Benchmarks/Middleware/FortunesDapperMiddleware.cs
@@ -2,15 +2,14 @@
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using System;
-using System.Collections.Generic;
 using System.Text.Encodings.Web;
 using System.Threading.Tasks;
 using Benchmarks.Configuration;
 using Benchmarks.Data;
+using Benchmarks.Templates;
 using Microsoft.AspNetCore.Builder;
 using Microsoft.AspNetCore.Http;
 using Microsoft.Extensions.DependencyInjection;
-using RazorSlices;
 
 namespace Benchmarks.Middleware
 {
@@ -20,13 +19,11 @@ namespace Benchmarks.Middleware
 
         private readonly RequestDelegate _next;
         private readonly HtmlEncoder _htmlEncoder;
-        private readonly SliceFactory<IEnumerable<FortuneUtf16>> _fortunesFactory;
 
         public FortunesDapperMiddleware(RequestDelegate next, HtmlEncoder htmlEncoder)
         {
             _next = next;
             _htmlEncoder = htmlEncoder;
-            _fortunesFactory = RazorSlice.ResolveSliceFactory<IEnumerable<FortuneUtf16>>("/Templates/FortunesUtf16.cshtml");
         }
 
         public async Task Invoke(HttpContext httpContext)
@@ -35,8 +32,9 @@ namespace Benchmarks.Middleware
             {
                 var db = httpContext.RequestServices.GetService<DapperDb>();
                 var rows = await db.LoadFortunesRows();
+                
 
-                await MiddlewareHelpers.RenderFortunesHtml(rows, httpContext, _htmlEncoder, _fortunesFactory);
+                await MiddlewareHelpers.RenderFortunesHtml(rows, httpContext, _htmlEncoder, FortunesUtf16.Create);
 
                 return;
             }

--- a/src/Benchmarks/Middleware/FortunesEfMiddleware.cs
+++ b/src/Benchmarks/Middleware/FortunesEfMiddleware.cs
@@ -2,7 +2,6 @@
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using System;
-using System.Collections.Generic;
 using System.Text.Encodings.Web;
 using System.Threading.Tasks;
 using Benchmarks.Configuration;
@@ -10,7 +9,6 @@ using Benchmarks.Data;
 using Microsoft.AspNetCore.Builder;
 using Microsoft.AspNetCore.Http;
 using Microsoft.Extensions.DependencyInjection;
-using RazorSlices;
 
 namespace Benchmarks.Middleware
 {
@@ -20,13 +18,11 @@ namespace Benchmarks.Middleware
 
         private readonly RequestDelegate _next;
         private readonly HtmlEncoder _htmlEncoder;
-        private readonly SliceFactory<IEnumerable<Fortune>> _fortunesFactory;
 
         public FortunesEfMiddleware(RequestDelegate next, HtmlEncoder htmlEncoder)
         {
             _next = next;
             _htmlEncoder = htmlEncoder;
-            _fortunesFactory = RazorSlice.ResolveSliceFactory<IEnumerable<Fortune>>("/Templates/Fortunes.cshtml");
         }
 
         public async Task Invoke(HttpContext httpContext)
@@ -36,7 +32,7 @@ namespace Benchmarks.Middleware
                 var db = httpContext.RequestServices.GetService<EfDb>();
                 var rows = await db.LoadFortunesRows();
 
-                await MiddlewareHelpers.RenderFortunesHtml(rows, httpContext, _htmlEncoder, _fortunesFactory);
+                await MiddlewareHelpers.RenderFortunesHtml(rows, httpContext, _htmlEncoder, Templates.Fortunes.Create);
 
                 return;
             }

--- a/src/Benchmarks/Middleware/FortunesRawMiddleware.cs
+++ b/src/Benchmarks/Middleware/FortunesRawMiddleware.cs
@@ -2,7 +2,6 @@
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using System;
-using System.Collections.Generic;
 using System.Text.Encodings.Web;
 using System.Threading.Tasks;
 using Benchmarks.Configuration;
@@ -10,7 +9,6 @@ using Benchmarks.Data;
 using Microsoft.AspNetCore.Builder;
 using Microsoft.AspNetCore.Http;
 using Microsoft.Extensions.DependencyInjection;
-using RazorSlices;
 
 namespace Benchmarks.Middleware
 {
@@ -20,13 +18,11 @@ namespace Benchmarks.Middleware
 
         private readonly RequestDelegate _next;
         private readonly HtmlEncoder _htmlEncoder;
-        private readonly SliceFactory<IEnumerable<Fortune>> _fortunesFactory;
 
         public FortunesRawMiddleware(RequestDelegate next, HtmlEncoder htmlEncoder)
         {
             _next = next;
             _htmlEncoder = htmlEncoder;
-            _fortunesFactory = RazorSlice.ResolveSliceFactory<IEnumerable<Fortune>>("/Templates/Fortunes.cshtml");
         }
 
         public async Task Invoke(HttpContext httpContext)
@@ -36,7 +32,7 @@ namespace Benchmarks.Middleware
                 var db = httpContext.RequestServices.GetService<RawDb>();
                 var rows = await db.LoadFortunesRows();
 
-                await MiddlewareHelpers.RenderFortunesHtml(rows, httpContext, _htmlEncoder, _fortunesFactory);
+                await MiddlewareHelpers.RenderFortunesHtml(rows, httpContext, _htmlEncoder, Templates.Fortunes.Create);
 
                 return;
             }

--- a/src/Benchmarks/Middleware/FortunesRawSyncMiddleware.cs
+++ b/src/Benchmarks/Middleware/FortunesRawSyncMiddleware.cs
@@ -2,7 +2,6 @@
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using System;
-using System.Collections.Generic;
 using System.Text.Encodings.Web;
 using System.Threading.Tasks;
 using Benchmarks.Configuration;
@@ -10,7 +9,6 @@ using Benchmarks.Data;
 using Microsoft.AspNetCore.Builder;
 using Microsoft.AspNetCore.Http;
 using Microsoft.Extensions.DependencyInjection;
-using RazorSlices;
 
 namespace Benchmarks.Middleware
 {
@@ -20,13 +18,11 @@ namespace Benchmarks.Middleware
 
         private readonly RequestDelegate _next;
         private readonly HtmlEncoder _htmlEncoder;
-        private readonly SliceFactory<IEnumerable<Fortune>> _fortunesFactory;
 
         public FortunesRawSyncMiddleware(RequestDelegate next, HtmlEncoder htmlEncoder)
         {
             _next = next;
             _htmlEncoder = htmlEncoder;
-            _fortunesFactory = RazorSlice.ResolveSliceFactory<IEnumerable<Fortune>>("/Templates/Fortunes.cshtml");
         }
 
         public Task Invoke(HttpContext httpContext)
@@ -36,7 +32,7 @@ namespace Benchmarks.Middleware
                 var db = httpContext.RequestServices.GetService<RawDb>();
                 var rows = db.LoadFortunesRowsSync();
 
-                return MiddlewareHelpers.RenderFortunesHtml(rows, httpContext, _htmlEncoder, _fortunesFactory);
+                return MiddlewareHelpers.RenderFortunesHtml(rows, httpContext, _htmlEncoder, Templates.Fortunes.Create);
             }
 
             return _next(httpContext);

--- a/src/Benchmarks/Middleware/MiddlewareHelpers.cs
+++ b/src/Benchmarks/Middleware/MiddlewareHelpers.cs
@@ -1,6 +1,7 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+using System;
 using System.Collections.Generic;
 using System.Text.Encodings.Web;
 using System.Threading.Tasks;
@@ -30,13 +31,13 @@ namespace Benchmarks.Middleware
         }
 
         public static async Task RenderFortunesHtml<T>(IEnumerable<T> model, HttpContext httpContext,
-            HtmlEncoder htmlEncoder, SliceFactory<IEnumerable<T>> fortunesFactory)
+            HtmlEncoder htmlEncoder, Func<IEnumerable<T>, RazorSlice> templateFactory)
         {
             httpContext.Response.StatusCode = StatusCodes.Status200OK;
             httpContext.Response.ContentType = "text/html; charset=UTF-8";
 
-            using var template = fortunesFactory(model);
-            await template.RenderToPipeWriterAsync(httpContext.Response.BodyWriter, htmlEncoder);
+            using var template = templateFactory(model);
+            await template.RenderAsync(httpContext.Response.BodyWriter, htmlEncoder);
             await httpContext.Response.BodyWriter.FlushAsync();
         }
     }

